index.js

```diff
@@ -1,72 +1,53 @@
-
-/**
- * Constantes
- */
-var RGBA = /rgba\s*\((\s*(\d+)\s*(,)\s*){3}(\s*(\d?\.\d+)\s*)\)$/i
-
-/**
- * PostCSS plugin to transform rgba() to hexadecimal
- */
-module.exports = postcss.plugin("postcss-color-rgba-fallback",
-function(options) {
-  options = options || {}
-
-  var properties = options.properties || [
-    "background-color",
-    "background",
-    "color",
-    "border",
-    "border-color",
-    "outline",
-    "outline-color",
-  ]
-
-  return function(style) {
-    style.walkDecls(function(decl) {
-      if (!decl.value ||
-          decl.value.indexOf("rgba") === -1 ||
-          properties.indexOf(decl.prop) === -1
-      ) {
-        return
-      }
-
-      // if previous prop equals current prop
-      // no need fallback
-      if (
-        decl.prev() &&
-        decl.prev().prop === decl.prop
-      ) {
-        return
-      }
-
-      var value = transformRgba(decl.value)
-      if (value) {
-        decl.cloneBefore({value: value})
-      }
-    })
-  }
-})
-
-/**
- * transform rgba() to hexadecimal.
- *
- * @param  {String} string declaration value
- * @return {String}        converted declaration value to hexadecimal
- */
-function transformRgba(string) {
-  var value = RGBA.exec(string)
-  if (!value) {
-    return
-  }
-
-  var rgb = colorString.getRgb(value[0])
-  var hex = colorString.hexString(rgb)
-  hex = string.replace(RGBA, hex)
-
-  return (hex)
-}
+var parser  = require('postcss-value-parser');
+var postcss = require('postcss');
+
+module.exports = postcss.plugin('postcss-color-rgba-fallback', function (opts) {
+	var method = opts && /^(copy|warn)$/.test(opts.method) ? opts.method : 'replace';
+	var props  = opts && opts.properties || ['background', 'background-color', 'color', 'border', 'border-bottom', 'border-bottom-color', 'border-color', 'border-left', 'border-left-color', 'border-right', 'border-right-color', 'border-top', 'border-top-color', 'outline', 'outline-color'];
+	var filter = opts && opts.filter;
+
+	return function (css, result) {
+		css.walkRules(function (rule) {
+			var cache = {};
+
+			rule.nodes.filter(function (decl) {
+				// assure property is cached
+				cache[decl.prop] = cache[decl.prop] || 0;
+
+				// count each time property is cached
+				++cache[decl.prop];
+
+				// keep valid, non-repeated, rgba-containing properties
+				return cache[decl.prop] === 1 && props.indexOf(decl.prop) !== -1 && /rgba/i.test(decl.value);
+			}).filter(function (decl) {
+				// keep uncaught non-repeated properties
+				return cache[decl.prop] === 1;
+			}).forEach(function (decl) {
+				// transpile rgba values
+				var value = parser(decl.value).walk(function (node) {
+					if (node.type === 'function' && node.value === 'rgba') {
+						var rgba = node.nodes.filter(function (word) {
+							return word.type === 'word';
+						}).map(function (word, index) {
+							return (0 + Math.round(word.value * (index > 2 ? 255 : 1)).toString(16)).slice(-2);
+						});
```

Comment:
It’s not fast, but it’s more *correct*. Still, I suppose we can make this presumption so I will change this.